src/step-function.ts

```diff
@@ -46,10 +46,34 @@ export type AnyStepFunction =
   | ExpressStepFunction<any, any>
   | StepFunction<any, any>;
 
+/**
+ * Machine and Execution context.
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/input-output-contextobject.html
+ *
+ * > Note: missing fields (ex: `$$.State.Name`) are abstracted away from {@link StepFunction}.
+ * > Note: `$$.Execution.Input` is available as the first parameter of the closure `(input) => {}`.
+ */
+export interface SfnContext {
+  readonly Execution: {
+    readonly Name: string;
+    readonly Id: string;
+    /**
+     * Format: ISO 8601
+     */
+    readonly StartTime: string;
+    readonly RoleArn: string;
+  };
+  readonly StateMachine: {
+    readonly Name: string;
+    readonly Id: string;
+  };
```

Comment:
Should we document each of the fields using the official docs as a reference? Also, perhaps have separate interface declarations instead of nested object literal types.

Reply:
we could, you thinking `context.executionName` `context.stateMachineName` and then map the parameters to their json path?

`executionName => $$.Execution.Name`